src/poll.c

```diff
@@ -34,6 +34,18 @@ static int luv_new_poll(lua_State* L) {
   return 1;
 }
 
+static int luv_new_socket_poll(lua_State* L) {
+  int fd = luaL_checkinteger(L, 1);
```

Comment:
Don't these survive being stored in ints as long as `sizeof(int) == sizeof(uintptr_t)`?  We use int for file descriptors in the fs apis.